server/controllers/profile.js

```diff
@@ -39,10 +49,24 @@ exports.createProfile = asyncHandler(async (req, res, next) => {
 // @route PUT /profiles/:id
 // @access Private
 exports.updateProfile = asyncHandler(async (req, res, next) => {
-  const profile = await Profile.findByIdAndUpdate(req.params.id, req.body, {
-    new: true,
-    runValidators: true,
-  });
+  const { firstName, lastName, description, gender, address, availability } =
+    req.body;
+
+  const profile = await Profile.findByIdAndUpdate(
+    req.params.id,
+    {
+      firstName,
+      lastName,
+      description,
+      gender,
+      address,
+      availability,
+    },
```

Comment:
Though you can directly pass the request body here for an update since it won't create any new attributes in the database but only update the existing ones. Additionally, you do not know ahead of time what attributes the client wants to update.

Reply:
I have this understanding for implementing create a profile by directly passing the request body as it won't create a new attribute in the database, even if a malicious user tries to do that.